<launch>

  <arg name="device_id" default="4"/>
  <arg name="master_id" default="3"/>
  <arg name="car_num" default="4"/>
  <arg name="drone_num" default="1"/>
  <arg name="frame" default="world"/>
  <arg name="fusionrate" default="50"/>
  <arg name="param_q" default="0.01"/>
  <arg name="param_r" default="30"/>
  <arg name="odom_topic" default="odom"/>

  <arg name="drone_id" default="4"/>
  <arg name="odometry_topic" default="visual_slam/odom"/>

  <node pkg="swarm_posefusion" name="posefusion_node" type="swarm_posefusion_node" output="screen">

    <param name="fusion/self_init" value="false"/>

    <param name="fusion/self_id" value="$(arg self_id)"/>
    <param name="fusion/uav_num" value="$(arg uav_num)"/>
    <param name="fusion/ugv_num" value="0"/>
    <param name="fusion/q" value="0.3"/>
    <param name="fusion/r" value="0.2"/>

    <remap from="~wheel_odom" to="$(arg origin_odom_topic)"/>
    <remap from="~fused_odom" to="$(arg fused_odom_topic)"/>

    <remap from="~uav_odom" to="$(arg uav_odom_topic)"/>
    <remap from="~ugv_odom" to="$(arg ugv_odom_receive_topic)"/>

    <!-- ************CREPES映射************ -->
    <remap from="~crepes/ugv_0_odom" to="fake_crepes/ugv_0_odom"/>
    <remap from="~crepes/ugv_1_odom" to="fake_crepes/ugv_1_odom"/>
    <remap from="~crepes/uav_0_odom" to="fake_crepes/uav_0_odom"/>
  </node>

  <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find swarm_posefusion)/config/rviz.rviz" required="true"/> -->
</launch>
